cnf-certification-test/faredge/suite.go

```diff
@@ -0,0 +1,127 @@
+
+package faredge
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/sirupsen/logrus"
+	"github.com/test-network-function/cnf-certification-test/cnf-certification-test/common"
+	"github.com/test-network-function/cnf-certification-test/cnf-certification-test/identifiers"
+	"github.com/test-network-function/cnf-certification-test/cnf-certification-test/results"
+	"github.com/test-network-function/cnf-certification-test/internal/clientsholder"
+	"github.com/test-network-function/cnf-certification-test/internal/crclient"
+	"github.com/test-network-function/cnf-certification-test/pkg/provider"
+	"github.com/test-network-function/cnf-certification-test/pkg/testhelper"
+)
+
+// All actual test code belongs below here.  Utilities belong above.
+var _ = ginkgo.Describe(common.FarEdgeTestKey, func() {
+	logrus.Debugf("Entering %s suite", common.FarEdgeTestKey)
+	var env provider.TestEnvironment
+	ginkgo.BeforeEach(func() {
+		env = provider.GetTestEnvironment()
+	})
+	ginkgo.ReportAfterEach(results.RecordResult)
+
+	testID, tags := identifiers.GetGinkgoTestIDAndLabels(identifiers.TestSchedOtherInSharedCpuPool)
+	ginkgo.It(testID, ginkgo.Label(tags...), func() {
+		nonGuranteedPods := env.GetNonGuaranteedPods()
+		testhelper.SkipIfEmptyAll(ginkgo.Skip, nonGuranteedPods)
+		testSchedOtherPolicyInSharedCPUPool(&env, filterNonGuaranteedContainers(nonGuranteedPods))
+	})
+})
+
+// Banashri - move inside filter code
+func filterNonGuaranteedContainers(pods []*provider.Pod) []*provider.Container {
+	var filteredContainers []*provider.Container
+	for _, pod := range pods {
+		filteredContainers = append(filteredContainers, pod.Containers...)
+	}
+	return filteredContainers
+}
+
+// Banashri
+func testSchedOtherPolicyInSharedCPUPool(env *provider.TestEnvironment, containers []*provider.Container) {
+	/*
+		Get the container ID
+		Run the command chrt -p <<container-process-id>>
+		Parse the output
+	*/
+	const (
+		ScheduleOtherPolicy   = "SCHED_OTHER"
+		ScheduleOtherPriority = 0
+	)
+
+	for _, container := range containers {
+		nodeName := container.NodeName
+		debugPod := env.DebugPods[nodeName]
+		if debugPod == nil {
+			ginkgo.Fail(fmt.Sprintf("Debug pod not found on Node: %s", nodeName))
+		}
+		ocpContext := clientsholder.NewContext(debugPod.Namespace, debugPod.Name, debugPod.Spec.Containers[0].Name)
+		pid, err := crclient.GetPidFromContainer(container, ocpContext)
+		if err != nil {
+			continue
+		}
+		logrus.Info(pid)
+		command := fmt.Sprintf("chrt -p %d", pid)
+
+		stdout, stderr, err := crclient.ExecCommandContainerNSEnter(command, container)
+		if err != nil || stderr != "" {
+			logrus.Error(err) // Add more stuffs
+		}
+
+		if !existsSchedulingPolicyAndPriority(stdout, ScheduleOtherPolicy, ScheduleOtherPriority) {
+			ginkgo.Fail("failed")
+		}
+	}
+
+	logrus.Println("Testing ended")
+}
+
+func existsSchedulingPolicyAndPriority(output, schedPolicy string, schedPriority int) bool {
```

Comment:
For organization, we could probably use a `scheduling` folder/file structure to hold these "extra" funcs in (and write unit tests for).  Same as we do in other suites.

Reply:
Done!